File: backend/app/schemas/permission.py
from pydantic import BaseModel, Field


class PermissionResponse(BaseModel):
    id: str
    code: str
    module: str
    label: str
    description: str | None = None

    model_config = {"from_attributes": True}


class RoleWithPermissionsResponse(BaseModel):
    id: str
    name: str
    description: str | None = None
    permissions: list[str] = []  # lista de codes


class UpdateRolePermissionsRequest(BaseModel):
    """Substitui a matriz da role pelo conjunto de codes informado."""
    permission_codes: list[str] = Field(default_factory=list)


class GenericMessage(BaseModel):
    message: str

File: frontend/src/types/permission.ts
export interface Permission {
  id: string
  code: string
  module: string
  label: string
  description: string | null
}

export interface RoleWithPermissions {
  id: string
  name: string
  description: string | null
  permissions: string[]
}

export interface UpdateRolePermissionsRequest {
  permission_codes: string[]
}
